### PR TITLE
Update internal tagging

### PR DESCRIPTION
**Pull request by xlucas**
### What does this PR do?

For consistency, let's update agent 6 internal tags to also contain the agent version, following #4330.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by albertvaka — DISMISSED** [review]

**Review by hush-hush — DISMISSED** [review]

**Comment by xlucas** [comment]
Postponing to 6.17

**Comment by xlucas** [comment]
Using this to clean up the pipeline a bit with recent changes introduced for agent 6 / 7.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -2117,14 +2063,14 @@ dev_branch_docker_hub:
...
-    - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-py2-jmx-amd64     datadog/agent-dev:${CI_COMMIT_REF_SLUG}-jmx
-    - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-py2-jmx-amd64     datadog/agent-dev:${CI_COMMIT_REF_SLUG}-py2-jmx
-    - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-py3-jmx-amd64     datadog/agent-dev:${CI_COMMIT_REF_SLUG}-py3-jmx
-    - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-py2py3-jmx-amd64  datadog/agent-dev:${CI_COMMIT_REF_SLUG}-py2py3-jmx
-    - inv -e docker.publish --signed-push ${SRC_DSD}:${SRC_TAG}-amd64               datadog/dogstatsd-dev:${CI_COMMIT_REF_SLUG}
+    - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-6-amd64             datadog/agent-dev:${CI_COMMIT_REF_SLUG}
+    - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-6-amd64             datadog/agent-dev:${CI_COMMIT_REF_SLUG}-py2
+    - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-7-amd64             datadog/agent-dev:${CI_COMMIT_REF_SLUG}-py3
+    - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-7-jmx-amd64         datadog/agent-dev:${CI_COMMIT_REF_SLUG}-jmx
```
Comment (KSerrania) [inline .gitlab-ci.yml]: I think it should be `6` here, as in all other places, the "default" image is a 6 image
```suggestion
    - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-6-jmx-amd64         datadog/agent-dev:${CI_COMMIT_REF_SLUG}-jmx
```

**Review by KSerrania — DISMISSED** [review]

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -2039,10 +1987,10 @@ twistlock_scan-6:
     - TWISTLOCK_PASS=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.twistlock_password --with-decryption --query "Parameter.Value" --out text)
     - scan () { echo -e "\n==== Scanning $1 ====\n"; docker pull $1 > /dev/null; /twistcli images scan --address="$TWISTLOCK_URL" --user="$TWISTLOCK_USER" --password="$TWISTLOCK_PASS" --vulnerability-threshold=$THRESHOLD --details $1; }
   script:
-    - scan ${SRC_AGENT}:${SRC_TAG}-py2-amd64
-    - scan ${SRC_AGENT}:${SRC_TAG}-py3-amd64
-    - scan ${SRC_AGENT}:${SRC_TAG}-py2-jmx-amd64
-    - scan ${SRC_AGENT}:${SRC_TAG}-py3-jmx-amd64
+    - scan ${SRC_AGENT}:${SRC_TAG}-6-amd64
+    - scan ${SRC_AGENT}:${SRC_TAG}-7-amd64
```
Comment (KSerrania) [inline .gitlab-ci.yml]: Nit: we already scan this image in `twistlock_scan-7`

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -2039,10 +1987,10 @@ twistlock_scan-6:
...
+    - scan ${SRC_AGENT}:${SRC_TAG}-6-jmx-amd64
+    - scan ${SRC_AGENT}:${SRC_TAG}-7-jmx-amd64
```
Comment (KSerrania) [inline .gitlab-ci.yml]: Nit: we already scan this image in `twistlock_scan-7`

File: .gitlab-ci.yml (hunk 2 above)
Comment (xlucas) [reply .gitlab-ci.yml]: Oh yeah, missed this

**Review by KSerrania — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]

**Pull request merged**